app/include/zmk/events/sensor_event.h

```diff
@@ -6,12 +6,21 @@
 
 #pragma once
 
-#include <zephyr/kernel.h>
-#include <zmk/event_manager.h>
 #include <zephyr/device.h>
+#include <zephyr/drivers/sensor.h>
+
+#include <zmk/event_manager.h>
+#include <zmk/sensors.h>
+
+// TODO: Move to Kconfig when we need more than one channel
+#define ZMK_SENSOR_EVENT_MAX_CHANNELS 1
+
 struct zmk_sensor_event {
-    uint8_t sensor_number;
-    const struct device *sensor;
+    uint8_t sensor_position;
```

Comment:
Sensors can be things that sense a change in physical position, so this name might be a bit confusing. If this is the sensor's index within the list of sensors, maybe call it an index instead of a position? If this is the virtual key position assigned to the sensor, maybe use `key_position` or `virtual_key_position`?

Also, are we actually saving any space by representing sensor positions as `uint8_t` instead of just `int`? At least in this struct, it's just going to get padded out to the size of an int anyways: https://godbolt.org/z/5YrKqPvKh

Reply:
Refactored to the terminology "sensor index" everywhere.

Storage in the struct will be padded, sure, but elsewhere for stack allocation, etc. it'll be lower. Also, if the event struct has more added later it might not actually grow, and can use some of the previous padding.

Reply:
> Also, if the event struct has more added later it might not actually grow, and can use some of the previous padding.

The compiler can't reorder structs, so you should place the smaller fields at the end if you want that to happen.

Reply:
Done.